### DISCUSSION TITLE
Exercise 10: modifications to the Chandy–Lamport snapshot protocol to make it strongly consistent snapshot

### DISCUSSION THREAD

**Discussion by dag7dev** [sapienzastudentsnetwork/distributed-systems#35] [Exercises]
**Exercise 10:** What modifications should be done to the Chandy–Lamport snapshot protocol so that it records a strongly consistent snapshot (i.e., all channel states are recorded empty).

**Comment by riccardoscuto** [comment]
From the Drive 

A cut $C$ is strongly consistent if, for all sending events $e \in C$ to another process, we also have the receiving event $e' \in C$. When a process receives its "take snapshot" message from the monitor, it broadcasts it without saving its local state before (like in the original Chandy-Lamport algorithm). The process will record its local state and send it to the monitor only when it receives its last "take snapshot" message from the other processes. Since the channels are FIFO, the last message sent from the other processes is always a "take snapshot" message, so no other kinds of messages can come after any process has recorded its local state.
